### DISCUSSION TITLE
ClickHouse Support

### DISCUSSION THREAD

**Discussion by r56473** [marimo-team/marimo#3303] [General]
"The SQL cell support is excellent—a great step towards bridging the gap between Python and SQL. With DuckDB as the underlying engine, the SQL cell functionality is essentially limited to what DuckDB supports. For example, I’d like to use SQL 'natively' with ClickHouse, but DuckDB does not support the ATTACH functionality required for that.

**Comment by akshayka** [comment]
Thanks for the discussion and the feedback. We are interested in adding "native" support for additional dialects of SQL, as well as different data sources.

While this isn't exactly what you asked for, you can follow along on progress here: https://github.com/marimo-team/marimo/issues/3201

In the meantime, are you able to use Clickhouse's Python client API?
